src/sqs/mod.rs

```diff
@@ -0,0 +1,313 @@
+use std::env;
+use std::rc::Rc;
+use std::time::Duration;
+use std::convert::TryFrom;
+
+use rocket::Config;
+use rusoto_core::Region;
+use rusoto_sqs::{DeleteMessageRequest, Message, ReceiveMessageRequest, Sqs, SqsClient};
+use serde_json;
+
+use error::Result;
+use failure::{self, err_msg};
+use logging::RBLogger;
+
+#[derive(Default, Debug, Clone)]
+pub struct SyncEvent {
+    pub event: String, // act on "delete", "device:delete"; ignore others
+    pub uid: String,
+    pub id: String,
+    pub handle: String,
+}
+
+impl TryFrom<Message> for SyncEvent {
+    type Error = failure::Error;
+
+    fn try_from(msg: Message) -> Result<Self> {
+        let body = match msg.body {
+            Some(d) => d,
+            None => {
+                return Err(err_msg("Missing body"))
+            }
+        };
+        let parsed: serde_json::Value = serde_json::from_str(&body).unwrap();
+        let blank = json!("");
+        let device = parsed.get("id").unwrap_or(&blank);
```

Comment:
unwrap_or_else which then makes the json! to avoid the work if it isn't needed.

Reply:
`std::ops::Fn<()>` not implemented for `serde::Value`, so looks like I'll need to stick with `unwrap_or` for now.

Reply:
yeah, can't do that. If I specify it as `unwrap_or_else(|| &json(""))` the reference lifespan is too short. Going to try something different.